### ISSUE TITLE
Code

### ISSUE THREAD

**Issue by sucessfullys** [xianggkl/VLU-Net#1]
Thank you very much for your work. Could you please provide the open-source code, or could you share an email address for further communication?

**Comment by xianggkl** [comment]
hi, I have uploaded the codes today (the pretrained ckpts will be uploaded tomorrow due to the poor web links, haha). If u have any questions, u can contact me by email: xmwang28@gmail.com.

**Comment by sucessfullys** [comment]
Thank you very much. I have already seen the code on GitHub and starred
the repository to show my support. I have a question: in the paper, are the
inputs x and y the same? From the code, they seem to be identical. I hope
you can clarify this for me. Thanks!

aya ***@***.***> 于2025年5月24日周六 17:23写道：

**Comment by xianggkl** [comment]
> Thank you very much. I have already seen the code on GitHub and starred
> the repository to show my support. I have a question: in the paper, are the
> inputs x and y the same? From the code, they seem to be identical. I hope
> you can clarify this for me. Thanks!
> 
> aya ***@***.***> 于2025年5月24日周六 17:23写道：
> […](#)

Inputs are the degraded images and the outputs are the clean images and they have the same shape.

**Comment by sucessfullys** [comment]
Sorry, I may not have expressed myself clearly. I was referring to the
'y' in the paper’s structural diagram."

aya ***@***.***> 于2025年5月28日周三 18:46写道：

**Comment by xianggkl** [comment]
> Sorry, I may not have expressed myself clearly. I was referring to the
> 'y' in the paper’s structural diagram."
> 
> aya ***@***.***> 于2025年5月28日周三 18:46写道：
> […](#)

oh，y is the degraded images which do not change as the degraded features, and x is the variable we aims to restore. specifically, x can be init by the degraded inputs.

**Comment by xianggkl** [comment]
though in traditional optimization steps, the x may usually be inited as zero, in deep networks x is often inited as the degraded inputs.

**Comment by sucessfullys** [comment]
Thank you very much for your reply.

aya ***@***.***> 于2025年5月28日周三 22:31写道：

**Comment by sucessfullys** [comment]
For the Clip fine-tuning, I only found the Jupyter notebook but not the
dataset processing code. Could you share that? I want to retrain the model.
Thanks.

duan yufa ***@***.***> 于2025年5月29日周四 14:51写道：

**Comment by xianggkl** [comment]
> For the Clip fine-tuning, I only found the Jupyter notebook but not the
> dataset processing code. Could you share that? I want to retrain the model.
> Thanks.
> 
> duan yufa ***@***.***> 于2025年5月29日周四 14:51写道：
> […](#)
Sorry for the uncomplete codes, I have updated the dataset processing codes (and more details can be found through open_clip due to we actually follow it).

**Comment by sucessfullys** [comment]
Thank you very much.

aya ***@***.***> 于2025年5月30日周五 00:05写道：

**Comment by sucessfullys** [comment]
May I ask if you experienced instability or fluctuations in the loss values
when training CLIP? Sometimes the loss values differ by as much as 2.
duan yufa ***@***.***> 于2025年5月30日周五 11:35写道：

**Comment by xianggkl** [comment]
pity that I kind of forgot the details, but I suggest that u can use other CLIP finetuning work as reference (like open_clip, DA_CLIP and so on). Also, visualize the score presented in Figure 5 of the paper can be a good way to value the performance.

**Comment by sucessfullys** [comment]
Ok, thank you very much.

aya ***@***.***> 于2025年5月30日周五 17:53写道：

**Comment by sucessfullys** [comment]
Sorry to bother you again, it seems I missed the import model_resume
statement. The model_resume functions, haha.

duan yufa ***@***.***> 于2025年5月30日周五 20:35写道：

**Comment by xianggkl** [comment]
these functions are basically for resume training from specific checkpoint, u can just remove or comment relevant codes.
